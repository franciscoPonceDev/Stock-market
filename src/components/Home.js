import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { FcStatistics } from 'react-icons/fc';
import Company from './Company';
import { fetchCompaniesFromAPI } from '../redux/home/home';
import './Company.scss';

const Home = () => {
  const companiesList = useSelector((state) => state.homeReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    if (companiesList.length === 0) {
      dispatch(fetchCompaniesFromAPI());
    }
  }, []);

  return (
    <section className="companies-container">
      <div className="company-item">
        <div className="company-header">
          <div className="title-icon">
            <FcStatistics className="icon" />
          </div>
          <h1>COMPANY&apos;S STOCK PERFORMANCE</h1>
        </div>
        <p className="company-subtitle">STATS BY COMPANY</p>
        <section className="company-section">
          {
            companiesList.map((company) => (
              <Company
                key={company.symbol}
                symbol={company.symbol}
                name={company.name}
                change={company.change}
                price={company.price}
                changesPercentage={company.changesPercentage}
              />
            ))
          }
        </section>
      </div>
    </section>
  );
};

export default Home;
