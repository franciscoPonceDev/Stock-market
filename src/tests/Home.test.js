import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from '../components/Home';

describe('Testing Home', () => {
  it('Render the Home page, should match to the snapshot', () => {
    const list = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>,
      )
      .toJSON();
    expect(list).toMatchSnapshot();
  });
});
