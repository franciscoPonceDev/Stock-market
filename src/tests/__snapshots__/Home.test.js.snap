// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Render the Home page, should match to the snapshot 1`] = `
<section
  className="companies-container"
>
  <div
    className="company-item"
  >
    <div
      className="company-header"
    >
      <div
        className="title-icon"
      >
        <svg
          className="icon"
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#37474F"
          >
            <rect
              height="36"
              width="2"
              x="23"
              y="5"
            />
            <rect
              height="14.2"
              transform="matrix(.707 .707 -.707 .707 33.523 -3.921)"
              width="2"
              x="20.5"
              y="31.4"
            />
            <rect
              height="14.2"
              transform="matrix(-.707 .707 -.707 -.707 72.487 46.995)"
              width="2"
              x="25.5"
              y="31.4"
            />
          </g>
          <rect
            fill="#CFD8DC"
            height="28"
            width="40"
            x="4"
            y="8"
          />
          <g
            fill="#607D8B"
          >
            <rect
              height="4"
              width="42"
              x="3"
              y="7"
            />
            <rect
              height="2"
              width="42"
              x="3"
              y="35"
            />
            <circle
              cx="31.5"
              cy="43.5"
              r="1.5"
            />
            <circle
              cx="16.5"
              cy="43.5"
              r="1.5"
            />
          </g>
          <g
            fill="#C51162"
          >
            <polygon
              points="31.9,18.9 26,24.9 20,18.9 11.9,26.9 14.1,29.1 20,23.1 26,29.1 34.1,21.1"
            />
            <polygon
              points="36,24 29,17 36,17"
            />
          </g>
        </svg>
      </div>
      <h1>
        COMPANY'S STOCK PERFORMANCE
      </h1>
    </div>
    <p
      className="company-subtitle"
    >
      STATS BY COMPANY
    </p>
    <section
      className="company-section"
    />
  </div>
</section>
`;
